save-orchestrator/src/main/kotlin/com/saveourtool/save/orchestrator/docker/DockerAgentRunner.kt

```diff
@@ -166,23 +164,23 @@ class DockerAgentRunner(
      * @throws RuntimeException if an exception not specific to docker has occurred
      */
     @Suppress("UnsafeCallOnNullableType", "TOO_LONG_FUNCTION")
-    private fun createContainerFromImage(baseImageTag: String,
-                                         pvId: DockerPvId,
-                                         workingDir: String,
-                                         runCmd: List<String>,
+    private fun createContainerFromImage(configuration: DockerService.RunConfiguration<DockerPvId>,
                                          containerName: String,
     ): String {
+        val baseImageTag = configuration.imageTag
+        val pvId = configuration.pvId
+        val runCmd = configuration.runCmd
         val envFileTargetPath = "$SAVE_AGENT_USER_HOME/.env"
         // createContainerCmd accepts image name, not id, so we retrieve it from tags
         val createContainerCmdResponse: CreateContainerResponse = dockerClient.createContainerCmd(baseImageTag)
-            .withWorkingDir(workingDir)
+            .withWorkingDir(EXECUTION_DIR)
             // Load environment variables required by save-agent and then run it.
             // Rely on `runCmd` format: last argument is parameter of the subshell.
             .withCmd(
                 // this part is like `sh -c` with probably some other flags
                 runCmd.dropLast(1) + (
                         // last element is an actual command that will be executed in a new shell
-                        "env \$(cat $envFileTargetPath | xargs) sh -c \"${runCmd.last()}\""
+                        """env $(cat $envFileTargetPath | xargs) sh -c "cp $SAVE_AGENT_USER_HOME/resources/* . && ${runCmd.last()}""""
```

Comment:
why we copy files from `/resources` to working dir?

Reply:
Now `resources` contains only three files: `save-agent.kexe`, `save-cli.kexe` and `.env`. Previousely we used to mount the volume with resources into working dir, because it contained everything that would be used during the execution. However, now each agent downloads resources for itself and we should either change where the volume is mounted and copy everything or provide CLI parameters with adjusted paths - I chose the former.